src/axelrod_dojo/archetypes/cycler.py

```diff
@@ -0,0 +1,142 @@
+import axelrod as axl
+from numpy import random
+
+from axelrod_dojo.utils import Params
+
+C, D = axl.Action
+
+
+class CyclerParams(Params):
+    """
+    Cycler params is a class to aid with the processes of calculating the best sequence of moves for any given set of
+    opponents. Each of the population in our algoruthem will be an instance of this class for putting into our
+    genetic lifecycle.
+
+    **********
+    TODOs:
+        * create a metaheuristic for mutations to increase mutation & crossover efficiency (ie, dont change the same
+        gene 2 times in a row or try and do a crossover if the genes being swapped are sufficiently different)
+    """
+
+    def __init__(self, sequence=None, sequence_length: int = 200, mutation_probability=0.1, mutation_potency=1):
+
+        if sequence is None:
+            # generates sequence uses a map to the Action class to change a sequence of 1 & 0s to C & D actions
+            self.sequence = self.generate_random_sequence(sequence_length)
+            self.sequence_length = sequence_length
+        else:
+            #  when passing a sequence, make a copy of the sequence to ensure mutation is for the instance only.
+            self.sequence = list(sequence)
+            self.sequence_length = len(sequence)
+
+        self.mutation_probability = mutation_probability
+        self.mutation_potency = mutation_potency
+
+    def __repr__(self):
+        return "{}".format(self.sequence)
+
+    @staticmethod
+    def generate_random_sequence(sequence_length):
+        """
+        Generates a sequence of random moves when an instance is initialised
+
+        Parameters
+        ----------
+        sequence_length - length or random moves to generate
+
+        Returns
+        -------
+        list - a list of C & D actions: e.g. [C,C,D,D,C]
+        """
+        return list(map(axl.Action, random.randint(0, 1 + 1, (sequence_length, 1))))
+
+    def crossover(self, other_cycler):
+        """
+        creates and returns a new CyclerParams instance with a single crossover point in the middle
+
+        Parameters
+        ----------
+        other_cycler - the other cycler where we get the other half of the sequence
+
+        Returns
+        -------
+        CyclerParams
+
+        """
+        seq1 = self.get_sequence()
+        seq2 = other_cycler.get_sequence()
+
+        midpoint = int(len(seq1) / 2)
+
+        # If the length is odd, keep more from sequence 1
```

Comment:
Does this introduce bias of some kind? Might be better to randomly choice which sequence to keep more of.

Reply:
any tips on an elegant way of testing a randomly generated split?

Reply:
Seeding the tests is the way to go :+1: (apologies if this comment is late or not helpful, connectivity is patchy at the moment so dipping in and out of things).